app/assets/javascripts/views/edit.js

```diff
@@ -0,0 +1,18 @@
+var editDocument = document.querySelectorAll('.edit_document')
+if (editDocument) {
+  document.querySelectorAll('[data-warn-before-unload=true]').forEach(function (element) {
+    element.addEventListener('change', function () {
+      if (!window.onbeforeunload) {
+        window.onbeforeunload = function () { return true }
```

Comment:
Do we have to do more than this for full browser support? 

I don't know if there are many browsers that still support custom text (maybe IE?) but I wonder if those ones may not show a message if this just returns true?

Also, any reason why we use `window.onbeforeunload` rather than `window.addEventListener` like we use for other events?